offlinedocs/pages/docs/[[...slug]].tsx

```diff
@@ -0,0 +1,545 @@
+
+  return {
+    props: {
+      content,
+      navigation,
+      route,
+      version,
+    },
+  }
+}
+
+const SidebarNavItem: React.FC<{ item: NavItem; nav: Nav }> = ({
+  item,
+  nav,
+}) => {
+  const router = useRouter()
+  let isActive = router.asPath.startsWith(`/docs/${item.path}`)
+
+  // Special case to handle the home path
+  if (item.path === "") {
+    isActive = router.asPath === "/docs/"
+
+    // Special case to handle the home path children
+    const homeNav = nav.find((navItem) => navItem.path === "") as NavItem
+    const homeNavPaths =
+      homeNav.children?.map((item) => `/docs/${item.path}/`) ?? []
+    if (homeNavPaths.includes(router.asPath)) {
+      isActive = true
+    }
+  }
+
+  return (
+    <Box>
+      <NextLink href={"/docs/" + item.path} passHref>
+        <Link
+          fontWeight={isActive ? 600 : 400}
+          color={isActive ? "gray.900" : "gray.700"}
+        >
+          {item.title}
+        </Link>
+      </NextLink>
+
+      {isActive && item.children && (
+        <Grid
+          as="nav"
+          pt={2}
+          pl={3}
+          maxW="sm"
+          autoFlow="row"
+          gap={2}
+          autoRows="min-content"
+        >
+          {item.children.map((subItem) => (
+            <SidebarNavItem key={subItem.path} item={subItem} nav={nav} />
+          ))}
+        </Grid>
+      )}
+    </Box>
+  )
+}
+
+const SidebarNav: React.FC<{ nav: Nav; version: string } & GridProps> = ({
+  nav,
+  version,
+  ...gridProps
+}) => {
+  return (
+    <Grid
+      h="100vh"
+      overflowY="scroll"
+      as="nav"
+      p={8}
+      w="300px"
+      autoFlow="row"
+      gap={2}
+      autoRows="min-content"
+      bgColor="white"
+      borderRightWidth={1}
+      borderColor="gray.200"
+      borderStyle="solid"
+      {...gridProps}
+    >
+      <Box mb={6}>
+        <Img src="/logo.svg" alt="Coder logo" />
+      </Box>
+
+      {nav.map((navItem) => (
+        <SidebarNavItem key={navItem.path} item={navItem} nav={nav} />
+      ))}
+    </Grid>
+  )
+}
+
+const MobileNavbar: React.FC<{ nav: Nav; version: string }> = ({
+  nav,
+  version,
+}) => {
+  const { isOpen, onOpen, onClose } = useDisclosure()
+
+  return (
+    <>
+      <Flex
+        bgColor="white"
+        px={6}
+        alignItems="center"
+        h={16}
+        borderBottomWidth={1}
+      >
+        <Img src="/logo.svg" alt="Coder logo" w={28} />
+
+        <Button variant="ghost" ml="auto" onClick={onOpen}>
+          <Icon as={MdMenu} fontSize="2xl" />
+        </Button>
+      </Flex>
+
+      <Drawer onClose={onClose} isOpen={isOpen}>
+        <DrawerOverlay />
+        <DrawerContent>
+          <DrawerCloseButton />
+          <DrawerBody p={0}>
+            <SidebarNav nav={nav} version={version} border={0} />
+          </DrawerBody>
+        </DrawerContent>
+      </Drawer>
+    </>
+  )
+}
+
+const slugifyTitle = (title: string) => {
+  return _.kebabCase(title.toLowerCase())
```

Comment:
IMO we should avoid the lodash dep if we're only using it for this function, you could inline the function here instead perhaps

Reply:
it is already been used for other packages tho, so no big deal IMO.